torch/jit/__init__.py

```diff
@@ -957,6 +957,18 @@ def _get_valid_constant(attr, v):
         """.format(type(v).__name__, attr, constants)))
 
 
+def _get_type(value):
```

Comment:
I am not a fan of adding this. It will need to be an exhaustive list, and we already have similar stuff in C++ for converting python values. It also cannot be done correctly: If a list is empty or a dict is empty we cannot infer the type. We should use the existing C++ code that figure out the type and IValue information from a python class rather than reimplement it here. We should also consider how we could inform the Module about the right type for things like empty dictionaries where the type is not inferrable.